src/librustdoc/json/conversions.rs

```diff
@@ -186,6 +188,7 @@ fn from_clean_item_kind(item: clean::ItemKind, tcx: TyCtxt<'_>) -> ItemEnum {
         PrimitiveItem(_) | KeywordItem(_) => {
             panic!("{:?} is not supported for JSON output", item)
         }
+        ExternCrateItem { .. } => unreachable!(),
```

Comment:
This will panic if you pass a `StrippedItem(ExternCrateItem)`. I would prefer to either move the match arm from above into this function (and add the `name` as a parameter), or to change the JSON format to match the internal format. It seems like a bug to mark `ExternCrateItem`s as not having a name anyway.

cc @HeroicKatora

Reply:
Just for the sake of my understanding, suppose that I have:
```
extern crate alloc as std;
```

Then the `name` is `alloc` but right now only `std` is accessible? I've ignored them for the documentation processor that I'm building but this would allow fully reconstructing the semantically correct statement.

Reply:
It's the other way around, right now you only have
```json
  "external_crates": {
    "1": {
      "name": "alloc",
      "html_root_url": "https://doc.rust-lang.org/nightly/"
    },
}
```
because rustdoc sees that the name is empty and doesn't include it in the main index.

Hmm, I guess that would change after this PR? @GuillaumeGomez can you see what the behavior is with `--output-format` json and add a test?

Reply:
Good idea!

Reply:
With this code:

```rust
pub extern crate alloc as whatever;
```

It doesn't add `whatever` to the `external_crates` object.

Reply:
Added the test to ensure there is no "whatever". ;)

Reply:
Sorry - I didn't mean `external_crates`, that will never have the name of the re-export. I meant in the main index. Do you see any item named `alloc` in the index now that wasn't there before?

Reply:
So after a discussion with @jyn514, it seems like there was no behavior change. All good then!